persistence/src/vespa/persistence/dummyimpl/dummy_bucket_executor.cpp

```diff
@@ -24,14 +24,14 @@ DummyBucketExecutor::~DummyBucketExecutor() {
 
 std::unique_ptr<BucketTask>
 DummyBucketExecutor::execute(const Bucket & bucket, std::unique_ptr<BucketTask> task) {
-    {
-        std::unique_lock guard(_lock);
-        while (_inFlight.contains(bucket.getBucket())) {
-            _cond.wait(guard);
-        }
-        _inFlight.insert(bucket.getBucket());
-    }
     _executor->execute(makeLambdaTask([this, bucket, bucketTask=std::move(task)]() {
+        {
```

Comment:
Why is the order here changed?

Reply:
This is just a simple implementation that wait if there are  tasks inflight. In the tests everything was in the same bucket. Then the test thread would block. And the test thread is the one that unblocks afterwards it can not block here. So I just moved the waiting to the executing thread.